Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/MaterialComponentController.cpp

```diff
@@ -186,11 +238,42 @@ namespace AZ
             for (auto& materialPair : m_configuration.m_materials)
             {
                 auto& materialAsset = materialPair.second.m_materialAsset;
-                if (materialAsset.GetId().IsValid() && !Data::AssetBus::MultiHandler::BusIsConnectedId(materialAsset.GetId()))
+
+                // This is a special case where a material was auto-generated from the model file, connected to a Material Component by the user,
+                // and then later a setting was changed to NOT auto-generate the model materials anymore. We need to switch to the new default
+                // material rather than trying to use the old default material which no longer exists. If that's the case, we reset the asset
+                // and OnModelReady will fill in the appropriate default material asset later.
                 {
-                    anyQueued = true;
-                    materialAsset.QueueLoad();
-                    Data::AssetBus::MultiHandler::BusConnect(materialAsset.GetId());
+                    Data::AssetId modelAssetId;
+                    MeshComponentRequestBus::EventResult(modelAssetId, m_entityId, &MeshComponentRequestBus::Events::GetModelAssetId);
+                    bool materialWasGeneratedFromModel = (modelAssetId.m_guid == materialAsset.GetId().m_guid);
+
+                    Data::AssetInfo assetInfo;
+                    Data::AssetCatalogRequestBus::BroadcastResult(assetInfo, &Data::AssetCatalogRequestBus::Events::GetAssetInfoById, materialAsset.GetId());
+                    bool materialAssetExists = assetInfo.m_assetId.IsValid();
+
+                    if (materialWasGeneratedFromModel && !materialAssetExists)
+                    {
+                        AZ_Warning("MaterialComponentController", false, "The default material assignment for this slot has changed and will be replaced (was '%s').",
+                            materialAsset.ToString<AZStd::string>().c_str());
+                        materialAsset.Reset();
+                    }
+                }
+
+                if (materialAsset.GetId().IsValid())
+                {
+                    if (!Data::AssetBus::MultiHandler::BusIsConnectedId(materialAsset.GetId()))
+                    {
+                        anyQueued = true;
+                        materialAsset.QueueLoad();
+                        Data::AssetBus::MultiHandler::BusConnect(materialAsset.GetId());
+                    }
+                }
+                else
+                {
+                    // Since a material asset wasn't found, we'll need to supply a default material. But the default materials
+                    // won't be known until after the mesh component has loaded the model data.
+                    MeshComponentNotificationBus::Handler::BusConnect(m_entityId);
```

Comment:
These changes are not necessary and should be reverted.  There is no way for the default material to be assigned unless the user did it explicitly.  If that asset was deleted or did not get generated because somebody changed a setting, the user should update their references.  The editor and inspector will warn about missing assets.  If we want this kind of fix up it should be a button or menu option in the editor component and not happen automatically at runtime in a game component.

Reply:
Yes, the user could set the default material explicitly. Currently, in order to set a material property override for the default FBX-converted material, the user has to manually assign that material to the override slot. This enables the possibility of setting property overrides on that material. As described in the code comment, this is the exact scenario I am addressing. Without these changes, the property override will be lost, and the controller will apparently be in an invalid state causing the preview icon to be incorrect (shows a page icon).

![image](https://user-images.githubusercontent.com/55155825/130279571-fc961ecd-e88a-4027-9f9e-120e4d5128bf.png)

Reply:
Explaining the scenario:
1. Material conversion is enabled.
2. Create a level with a model. User wants to change the color but doesn't need an editable material. Manually assign the generated default material to the slot. This exposes the Edit Material Instance option so you can change the color to red. When you save, it saves the AssetId of the default material as an override, and the red property override.
3. Shut everything down and disable material conversion. When you run the AP again, it will remove all the old generated materials.
4. Open the level. It tries to load the material override, but that material no longer exists. You get the screenshot I posted on GitHub.

Reply:
We discussed offline and decided it's better revert this messy fix to an obscure edge case and instead have a separate task to generally address the case of missing material assets.